src/core/mavlink_parameters.cpp

```diff
@@ -363,6 +363,14 @@ void MAVLinkParameters::process_param_ext_value(const mavlink_message_t& message
                 if (work->get_param_callback) {
                     work->get_param_callback(MAVLinkParameters::Result::SUCCESS, value);
                 }
+            } else if (value.is_uint8() && work->param_value.is_uint16()) {
+                // FIXME: workaround for mismatching type uint8_t which should be uint16_t.
+                ParamValue correct_type_value;
+                correct_type_value.set_uint16(uint16_t(value.get_uint8()));
```

Comment:
Shouldn't it be a `static_cast` in C++11?

Reply:
Would be nicer yes.